### PR TITLE
[MFTF] Refactoring of AdminAvailabilityCreditMemoWithNoPaymentTest

### LINKED ISSUE #30829: [Issue] WIP: [MFTF] Refactoring of AdminAvailabilityCreditMemoWithNoPaymentTest
This issue is automatically created based on existing pull request: magento/magento2#30792: WIP: [MFTF] Refactoring of AdminAvailabilityCreditMemoWithNoPaymentTest

---------
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Go to Admin
2. Start creating a new order using the Create New Customer functionality
3. Add a product to the Order
4. Adjust product price to 0 (via the Custom Price)
5. Place the Order
6. Invoice the Order
7. Check if it is possible to create the Credit Memo

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Go to Admin
2. Start creating a new order using the Create New Customer functionality
3. Add a product to the Order
4. Adjust product price to 0 (via the Custom Price)
5. Place the Order
6. Invoice the Order
7. Check if it is possible to create the Credit Memo

### Resolved issues:
1. [x] resolves magento/magento2#30829: WIP: [MFTF] Refactoring of AdminAvailabilityCreditMemoWithNoPaymentTest

### CODE REVIEW

**Comment by sidolov** [comment]
@magento create issue

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AdminSetCustomPriceActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminSetCustomPriceActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Sales/Test/Mftf/ActionGroup/AdminSetCustomPriceActionGroup.xml]: I would suggest providing more detailed naming for the action groups. So, the name of the action group describes its purpose in detail. For example, the action group above `AdminFillAccountInformationOnCreateOrderPageActionGroup` ideally describes the case. So we know where exactly we fill the account information. With `AdminSetCustomPriceActionGroup` it's not obvious where it's reusable. 

Hope it makes sense.

Thanks

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AdminSetCustomPriceActionGroup.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Sales/Test/Mftf/ActionGroup/AdminSetCustomPriceActionGroup.xml]: @rogyar naming has been adjusted, thank you!

**Review by rogyar — APPROVED** [review]

**Comment by eduard13** [comment]
Hi @AnnaAPak, could you please make sure you properly merge the mainline?

Thank you.

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B, Semantic Version Checker

**Comment by AnnaAPak** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Bravo** [comment]
Hi @AnnaAPak,

As per this https://github.com/magento/magento2/issues/30829#issuecomment-2011301588 comment since issue is no more reproducible.We are Closing this PR.

Please feel free to reopen the PR.

Thanks.

**Pull request closed without merging**